### DISCUSSION TITLE
Testwatcher facility?

### DISCUSSION THREAD

**Discussion by steckerhalter** [getgauge/gauge#1897] [General]
Hi... I'm using Gauge with Kotlin at the moment.

Previously I have been using JUnit5 where I could execute code for example when failure occurred: https://junit.org/junit5/docs/5.5.1/api/org/junit/jupiter/api/extension/TestWatcher.html

Is there a similar facility in Gauge I could use?

Thank you.

**Comment by zabil** [comment]
Gauge does not have this feature. 

As a workaround you could add a script to detect a failure exit code after a Gauge run to trigger another program. You could also use the xml/json reporting plugin to parse the result and detect failures.

**Reply by steckerhalter** [reply]
Thanks I'm using Selenoid to record videos of the tests and therefore need the Selenium context to extract the video url. I'll see what I can do. Maybe I can hook into the Java code somehow but it doesn't look easy.

Perhaps Gauge could add a feature for this with annotations in the future? e.g.

```java
@AfterFailure

@AfterSuccess

@AfterAbortion
```

or something similar?

**Reply by zabil** [reply]
You can propose this as a feature at https://github.com/getgauge/gauge/issues.
To be honest it's a lot of work as any new feature need to be ported to all langauge runners. At the moment there are no plans to add new features to Gauge. A pull request to support this will be nice.

**Reply by steckerhalter** [reply]
ok thanks
